src/AbstractSQLOptimiser.ts

```diff
@@ -1025,13 +1035,13 @@ const typeRules = {
 						return;
 					}
 				} else if (arg[0] === 'NotIn') {
-					const fieldRef = arg[1] as string;
```

Comment:
Was this really a string?
B/c `NotInNode[1]` gives `FieldNode | ReferencedFieldNode` !?!?!?
![image](https://github.com/user-attachments/assets/13b29167-e118-4e2a-b6ac-de127fe9ac6b)

Reply:
It was not a string, but it would have been automatically coerced into a string when used to index the object - this is almost certainly from having previously been ometajs and mistyped when converting to typescript

Reply:
Or maybe it *was* a string when first typed but then other changes happened, either way it's typed better now